pkg/dashboard/resource/function.go

```diff
@@ -404,19 +404,18 @@ func (fr *functionResource) validateUpdateInfo(functionInfo *functionInfo, funct
 }
 
 func (fr *functionResource) processFunctionInfo(functionInfoInstance *functionInfo, projectName string) (*functionInfo, error) {
-
-	// override namespace if applicable
-	if functionInfoInstance.Meta != nil {
-		functionInfoInstance.Meta.Namespace = fr.getNamespaceOrDefault(functionInfoInstance.Meta.Namespace)
+	if functionInfoInstance.Meta == nil {
+		functionInfoInstance.Meta = &functionconfig.Meta{}
 	}
 
-	// meta must exist
-	if functionInfoInstance.Meta == nil ||
-		functionInfoInstance.Meta.Name == "" ||
-		functionInfoInstance.Meta.Namespace == "" {
-		err := errors.New("Function name must be provided in metadata")
+	functionInfoInstance.Meta.Namespace = fr.getNamespaceOrDefault(functionInfoInstance.Meta.Namespace)
+
+	if functionInfoInstance.Meta.Name == "" {
+		return nil, nuclio.NewErrBadRequest("Function name must be provided in metadata")
+	}
 
-		return nil, nuclio.WrapErrBadRequest(err)
+	if functionInfoInstance.Meta.Namespace == "" {
```

Comment:
well, there are some unittests behavior that verify that and they fail on a case where
1. function request does not contain namespace
2. resolved namespace from server is empty

I will leave this sanity validation, to avoid handling another possible bug and address it on another pr